src/cockpit/samples.py

```diff
@@ -25,6 +26,15 @@
 MS_PER_JIFFY = 1000 / (USER_HZ if (USER_HZ > 0) else 100)
 
 
+@contextlib.contextmanager
+def os_open(*args, **kwargs):
+    fd = os.open(*args, **kwargs)
+    try:
+        yield fd
+    finally:
+        os.close(fd)
+
+
```

Comment:
In my experience, generators can be a bit slow.

The only thing you ever do with this is to call os.read() on it, so why not just make the function return the bytes?

Reply:
Well, the generator does a bunch more, we catch file found exceptions and value errors and it is the most pythonic way to write it I'd say.

We could measure it for sure.